### PR TITLE
IBP-3899 Refactor Sorted Page Request

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/dao/dms/DmsProjectDao.java [hunk 1]
```diff
@@ -1457,11 +1458,11 @@ public long countStudies(final StudySearchFilter studySearchFilter) {
 		return ((BigInteger) sqlQuery.uniqueResult()).longValue();
 	}
 
-	public List<StudyDto> getStudies(final StudySearchFilter studySearchFilter) {
+	public List<StudyDto> getStudies(final StudySearchFilter studySearchFilter, final PageRequest pageRequest) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/dao/dms/DmsProjectDao.java:1461]: Better use the `final Pageable pageable` in this DAO class to be consistent with other classes

File: src/main/java/org/generationcp/middleware/dao/dms/ObservationUnitsSearchDao.java [hunk 2]
```diff
@@ -1077,17 +1079,17 @@ private String getEnvironmentColumnName(final String variableName) {
 	}
 
 	private List<Map<String, Object>> getObservationUnitTableAsMapListResult(final ObservationUnitsSearchDTO searchDto,
-		final String observationVariableName) {
+		final String observationVariableName, final PageRequest pageRequest) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/dao/dms/ObservationUnitsSearchDao.java:1082]: Same here use `final Pageable pageable` to be consistent

File: src/main/java/org/generationcp/middleware/service/api/dataset/DatasetService.java [hunk 3]
```diff
@@ -219,7 +222,7 @@ long countFilteredObservationUnitsForDataset(Integer datasetId, Integer instance
 	 * @return List of Variable (Column) Name and Value Map
 	 */
 	List<Map<String, Object>> getObservationUnitRowsAsMapList(
-		int studyId, int datasetId, ObservationUnitsSearchDTO searchDTO);
+		int studyId, int datasetId, ObservationUnitsSearchDTO searchDTO, PageRequest pageRequest);
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/api/dataset/DatasetService.java:225]: Use final Pageable pageable to be consistent

File: src/main/java/org/generationcp/middleware/service/api/study/StudyService.java [hunk 4]
```diff
@@ -74,7 +75,7 @@
 
 	Integer getEnvironmentDatasetId(int studyId);
 
-	List<StudyDto> getStudies(StudySearchFilter studySearchFilter);
+	List<StudyDto> getStudies(StudySearchFilter studySearchFilter, PageRequest pageRequest);
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/api/study/StudyService.java:78]: Use` final Pageable pageable`

File: src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java [hunk 5]
```diff
@@ -929,7 +922,9 @@ public void setValueToVariable(final Integer datasetId, final ObservationUnitsPa
 		final String newValue = paramDTO.getNewValue();
 		final String variableId = paramDTO.getObservationUnitsSearchDTO().getFilter().getVariableId().toString();
 		final List<Phenotype> phenotypes = new ArrayList<>();
-		this.fillSearchDTO(studyId, datasetId, paramDTO.getObservationUnitsSearchDTO());
+
+		// TODO: IBP-3899
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java:926]: Remove TODO

File: src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java [hunk 6]
```diff
@@ -579,31 +580,23 @@ protected void setDaoFactory(final DaoFactory daoFactory) {
...
-	private void fillSearchDTO(final int studyId, final int datasetId, final ObservationUnitsSearchDTO searchDTO) {
-		if (searchDTO.getSortedRequest() != null && searchDTO.getSortedRequest().getSortBy() != null) {
-			final Term term = this.ontologyDataManager.getTermById(Integer.valueOf(searchDTO.getSortedRequest().getSortBy()));
-			if (term != null) {
-				searchDTO.getSortedRequest()
-					.setSortBy(term.getName());
-			}
-		}
-
+	private void fillSearchDTO(final int studyId, final int datasetId, final ObservationUnitsSearchDTO searchDTO, final Pageable pageable) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java:599]: pageable parameter is unused in `fillSearchDTO`

File: src/main/java/org/generationcp/middleware/service/impl/inventory/PlantingServiceImpl.java [hunk 7]
```diff
@@ -48,8 +57,9 @@ public PlantingPreparationDTO searchPlantingPreparation(
 		this.processSearchComposite(searchDTO);
 
 		// observation units
+		// TODO: IBP-3899
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/impl/inventory/PlantingServiceImpl.java:60]: Why is there a TODO here still?

File: src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java [hunk 8]
```diff
@@ -607,8 +608,8 @@ public long countPhenotypes(final PhenotypeSearchRequestDTO requestDTO) {
 	}
 
 	@Override
-	public List<StudyDto> getStudies(final StudySearchFilter studySearchFilter) {
-		return this.daoFactory.getDmsProjectDAO().getStudies(studySearchFilter);
+	public List<StudyDto> getStudies(final StudySearchFilter studySearchFilter, final PageRequest pageRequest) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java:611]: Same here, use `final Pageable pageable`

**Review by darla-leafnode — CHANGES_REQUESTED** [review]
Hi @abatac - please address my comments

**Comment by abatac** [comment]
> Hi @abatac - please address my comments

This is done.

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**